src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/MvcViewOptionsSetup.cs

```diff
@@ -13,24 +13,29 @@ namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
     /// <summary>
     /// Sets up default options for <see cref="MvcViewOptions"/>.
     /// </summary>
-    public class MvcViewOptionsSetup : ConfigureOptions<MvcViewOptions>
+    public class MvcViewOptionsSetup : IConfigureOptions<MvcViewOptions>
     {
+        private IServiceProvider _serviceProvider;
+
         /// <summary>
         /// Initializes a new instance of <see cref="MvcViewOptionsSetup"/>.
         /// </summary>
         public MvcViewOptionsSetup(IServiceProvider serviceProvider)
-            : base(options => ConfigureMvc(options, serviceProvider))
         {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            _serviceProvider = serviceProvider;
```

Comment:
Was there a reason we couldn't use the same pattern as `MvcDataAnnotationsMvcOptionsSetup` here?